src/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs

```diff
@@ -259,22 +263,14 @@ private static MethodDefinitionHandle BasicValidationEmit(MetadataBuilder metada
 
             // .endfinally
             il.OpCode(ILOpCode.Endfinally);
-            int handlerEnd = il.Offset;
-
-            // END: 
-            il.MarkLabel(endLabel);
+            il.MarkLabel(finallyEnd);
 
             // ret
             il.OpCode(ILOpCode.Ret);
 
-            var body = methodBodies.AddMethodBody(exceptionRegionCount: 1);
-            var eh = body.WriteInstructions(codeBuilder, branchBuilder, out mainBodyOffset);
-            eh.StartRegions();
-            eh.AddFinally(tryOffset, handlerOffset - tryOffset, handlerOffset, handlerEnd - handlerOffset);
-            eh.EndRegions();
-
+            int mainBodyOffset = methodBodyStream.AddMethodBody(il);
             codeBuilder.Clear();
-            branchBuilder.Clear();
+            flowBuilder.Clear();
 
```

Comment:
The exception handling with labels via ControlFlowBuilder is really nice vs. what you had to do before! :)